minetest/minetest
# Add setting for improved 3d sounds with headphones (HRTF)

[Head-related Transfer Function](https://en.wikipedia.org/wiki/Head-related_transfer_function) is a technique that improves the quality of 3d sounds when wearing headphones. Rather than just changing stereo balance/volume, it also changes the time offset. This results in sounds being much clearer and easier to locate.

**See this video for the difference: https://www.youtube.com/watch?v=jXkDmJRI1yg**

OpenAL is supposed to automatically enable HRTF when it detects headphones. Unfortunately, it's unable to do this on pipewire, or ALSA, and it's unreliable on Windows or pulseaudio. (Citation: #openal on liberachat).

It's possible to enable it system wide by adding `htrf = true` to `~/.alsoftrc` but this isn't discoverable currently.

The PR adds a setting that allows the user to force enable this feature. With the settings redesign, this become a lot more prominent.

This PR also includes #12628 for testing reasons

### Alternatives

With the settings redesign, we could add a button to a help page telling users how to enable HRTF system wide, rather than just for Minetest

## To do

This PR is a Work in Progress

- [ ] Add support for changing at run time. This is possible, but I'm unsure of where to actually hook this in

## How to test

Wear headphones and use `/test3dsound` in devtest.

Also check whether the "Audio: Global Initialized: OpenAL" log says HRTF is enabled or not
